pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -232,6 +245,27 @@ def _configure_fault_tolerance_callbacks(self):
         # don't use `log_dir` to minimize the chances of failure
         self.trainer.callbacks.append(_FaultToleranceCheckpoint(dirpath=self.trainer.default_root_dir))
 
+    def _configure_external_callbacks(self) -> None:
+        """Add external callbacks registered through entry points.
+
+        The entry points are expected to be functions returning a list of callbacks, which will be added to the Trainer
+        callback list.
+        """
+        if _PYTHON_GREATER_EQUAL_3_8_0:
+            factories = importlib.metadata.entry_points().get("pytorch_lightning.callbacks_factory", ())
+        else:
+            factories = iter_entry_points("pytorch_lightning.callbacks_factory")
+
+        for factory in factories:
+            callback_factory = factory.load()
+            callbacks_list: List[Callback] = callback_factory()
+            callbacks_list = [callbacks_list] if isinstance(callbacks_list, Callback) else callbacks_list
+            _log.info(
+                f"Adding {len(callbacks_list)} callbacks from entry point '{factory.name}':"
+                f" {', '.join(type(cb).__name__ for cb in callbacks_list)}"
+            )
+            self.trainer.callbacks.extend(callbacks_list)
```

Comment:
```suggestion
            callbacks_list = [callbacks_list] if isinstance(callbacks_list, Callback) else callbacks_list
            if not isinstance(callbacks_list, Iterable):
                raise TypeError(f"Expected callbacks from entry point '{factory.name}' to be iterable, but received this instead: {type(callbacks_list)}.")
            for callback in callbacks_list:
                if not isinstance(callback, Callback):
                    raise TypeError(f"Expected callback from entry point '{factory.name}' to be a Callback, but received this instead: {type(callback)}.")
            self.trainer.callbacks.append(callback)
            _log.info(
                f"Adding {len(callbacks_list)} callbacks from entry point '{factory.name}':"
                f" {', '.join(type(cb).__name__ for cb in callbacks_list)}"
            )
```

Reply:
this validation is what I had a few commits ago, @carmocca suggested to remove it, @tchaton voted to keep it. 
https://github.com/PyTorchLightning/pytorch-lightning/pull/12739#discussion_r850293743
I can add it back